config/identify_system.go

```diff
@@ -46,6 +46,11 @@ func identifySystem(config ServerConfig) (systemType string, systemScope string,
 		if systemID == "" {
 			systemID = config.CrunchyBridgeClusterID
 		}
+	} else if (config.AivenServiceID != "") || systemType == "aiven" {
+		systemType = "aiven"
+		if systemID == "" {
+			systemID = config.AivenServiceID
```

Comment:
I wonder if we should be setting something like `systemTypeFallback` and `systemIDFallback` here.

For comparison, we have similar logic earlier in this file for AWS, since we changed how we identify AWS systems, and we use the fallback logic to make sure that the API first checks if there is an existing server that it should lookup based on the fallback values, and only then would go ahead and do the regular logic.

That fallback logic doesn't support the ID and Type fields yet, but there is no reason it couldn't. And that way we would have a seamless transition for servers that are currently identified as "Self-managed" but are actually Aiven. Otherwise upgrading the collector would cause them to get a new server entry in pganalyze (with all new statistics going to that new server entry).

Reply:
That's a good point. I copied the Crunchy integration here, but you're right, this case is different. I'll look into that.

Reply:
Yep, and I just tested this and it worked as expected :)

Reply:
Great--I ended up doing that as well, and it worked too.